src/osp/Resource/blueprints.h

```diff
@@ -50,16 +54,22 @@ struct BlueprintMachine
 struct BlueprintPart
 { 
 
-    uint32_t m_partIndex; // index to BlueprintVehicle's m_partsUsed
+    BlueprintPart(uint32_t protoIndex, uint16_t machineCount,
+                  Vector3 translation, Quaternion rotation, Vector3 scale)
+    : m_protoIndex(protoIndex)
+    , m_translation(translation)
+    , m_rotation(rotation)
+    , m_scale(scale)
+    , m_machineCount(machineCount)
+    { }
+
+    uint32_t m_protoIndex; // index to BlueprintVehicle's m_partsUsed
 
     Vector3 m_translation;
     Quaternion m_rotation;
     Vector3 m_scale;
 
-    // put some sort of config here
-
-    // Configuration of individual machines
-    std::vector<BlueprintMachine> m_machines;
+    uint16_t m_machineCount;
```

Comment:
Only 65536 machines?

Reply:
This is the maximum number of machines that can be added to a single part. It is rather unlikely that this many machines are ever needed on a single part.